src/renderer/actions/install.js

```diff
@@ -129,59 +147,141 @@ export default async function(config) {
     const sane = doSanityCheck(config);
     if (!sane) return fail();
 
-
+    // Installation channels (such as "stable", "canary") and their latest version's paths.
     const channels = Object.keys(config);
-    const paths = Object.values(config);
+    const discordPaths = Object.values(config);
+
 
+    // Legacy BD paths. We will offer to migrate old data from these locations.
+    const legacyBDFolders = [];
+    if (process.platform === "darwin") {  // Mac.
+        legacyBDFolders.push(path.join(remote.app.getPath("home"), "Library", "Preferences", "betterdiscord"));
+    }
+    else if (process.platform === "linux") {
+        // Example: Migrating data from `~/.config/BetterDiscord` to a new location.
+        // NOTE: This is just an example if we decide to move to per-Discord BD installations later.
+        //legacyBDFolders.push(path.join(remote.app.getPath("appData"), "BetterDiscord"));
+    }
+
+    // Latest BD paths. This is where we'll install BetterDiscord's ASAR package.
+    const bdFolders = [];
+    if (process.platform === "win32" || process.platform === "darwin") {
+        // Windows and Mac: BetterDiscord stored in a single, shared location.
+        bdFolders.push(path.join(remote.app.getPath("appData"), "BetterDiscord"));
+    }
+    else {  // Linux.
+        // NOTE: On Linux, we install BetterDiscord into every Discord instance's
+        // individual xdg-config directory, to be able to support Flatpaks and Snaps,
+        // and to avoid conflicts if multiple sandboxed Discord versions run at the
+        // same time. Only the sandboxes have truly unique "per-installation" BD paths.
+        for (const discordPath of discordPaths) {
+            // Jump out of the latest Discord version's subdirectory, to the Discord
+            // installation's main xdg-config directory, and install BetterDiscord there.
+            // Before: "/home/foo/.var/app/com.discordapp.Discord/config/discord/0.0.19/modules/discord_desktop_core".
+            // After: "/home/foo/.var/app/com.discordapp.Discord/config/BetterDiscord".
+            // NOTE: Installation paths on Linux always contain those 4 suffix parts.
+            // NOTE: The requirement that BetterDiscord is installed in xdg-config
+            // is hardcoded into betterdiscord.asar, so we can't change these paths.
```

Comment:
Well BetterDiscord itself (and the corresponding asar) can be changed in future releases if it would make things like this less janky. This feels very breakable.

Reply:
I would love to work together on that. The best solution would be if BetterDiscord scans a subdirectory of the currently running Discord's config directory. Perhaps by checking both the old (`~/.config/BetterDiscord`) and new paths, and preferring the new one, which would be "Directly under the parent Discord version's config directory", like this:

`~/.config/discord/BetterDiscord`.

Having such a structure would allow multiple native versions of Discord to co-exist. If a person installs Canary and Stable and PTB all on the same machine, they wouldn't overwrite each other's configs if all 3 are running at the same time.

Currently, Snap and Flatpak versions don't have that issue, since those are already sandboxed into individual config directories.

As for whether the current code you're commenting on is janky or breakable: It's not, because the installation path is determined by us in the same script: BetterDiscord itself searches for its own config in the config-directory, which is always 4 steps above the Discord path. We're the ones writing BetterDiscord into a path that is exactly 4 steps above the Discord path. So it's totally reliable. It's just a relative path to avoid having hardcoded paths (since the *OLD* hardcoded paths *WERE* very breakable).

Reply:
Please reach out, I would like to learn the best path forward on this.